test/Dialect/ESI/cosim.mlir

```diff
@@ -27,9 +28,27 @@ module {
     // CAPNP-LABEL: struct TYsi14 @0x8ee0bd493e80e8a1
     // CAPNP:         i @0 :Int16;
 
-    // COSIM: %rawOutput, %valid = esi.unwrap.vr %send.x, %TestEP.DataInReady : si14
-    // COSIM: %0 = esi.encode.capnp %rawOutput : si14 -> !rtl.array<192xi1>
-    // COSIM: %TestEP.DataOutValid, %TestEP.DataOut, %TestEP.DataInReady = rtl.instance "TestEP" @Cosim_Endpoint(%clk, %rstn, %ready, %valid, %0) {parameters = {ENDPOINT_ID = 1 : i32, RECVTYPE_SIZE_BITS = 192 : i64, RECV_TYPE_ID = 14000240888948784983 : ui64, SEND_TYPE_ID = 10295436870447851681 : ui64, SEND_TYPE_SIZE_BITS = 192 : i64}} : (i1, i1, i1, i1, !rtl.array<192xi1>) -> (i1, !rtl.array<192xi1>, i1)
-    // COSIM: %1 = esi.decode.capnp %TestEP.DataOut : !rtl.array<192xi1> -> !esi.channel<i32>
+    // COSIM: rtl.instance "TestEP" @Cosim_Endpoint(%clk, %rstn, %{{.*}}, %{{.*}}, %{{.*}}) {parameters = {ENDPOINT_ID = 1 : i32, RECV_TYPE_ID = 14000240888948784983 : ui64, RECV_TYPE_SIZE_BITS = 128 : i32, SEND_TYPE_ID = 10295436870447851681 : ui64, SEND_TYPE_SIZE_BITS = 128 : i32}} : (i1, i1, i1, i1, i128) -> (i1, !rtl.array<128xi1>, i1)
+
+    // SV: Reciever recv (
+    // SV:   .a ({{.*}}.source)
+    // SV: );
+    // SV: {{.*}}.ready = TestEP_DataInReady;
+    // SV: Sender send (
+    // SV:   .x ({{.*}}.sink)
+    // SV: );
+    // SV: Cosim_Endpoint #(.ENDPOINT_ID(32'd1), .RECV_TYPE_ID(64'd14000240888948784983), .RECV_TYPE_SIZE_BITS(32'd128), .SEND_TYPE_ID(64'd10295436870447851681), .SEND_TYPE_SIZE_BITS(32'd128)) TestEP (        // <stdin>:42:66
+    // SV:   .clk (clk),
+    // SV:   .rstn (rstn),
+    // SV:   .DataOutReady ({{.*}}.ready),
+    // SV:   .DataInValid ({{.*}}.valid),
+    // SV:   .DataIn ({{[{]}}{50'h0, _T_1}, {16'h0, 16'h1, 32'h0}}),
```

Comment:
I think it might be slightly preferable to bind a name for these temp variables with some earlier `// SV: ` checks, and use the bound name here instead of referring to it directly. That would prevent the test from breaking if the temp variable naming scheme ever changes.

On the other hand, then you would also be testing the creation of wires for these temps, which is kind of unrelated to this test, and could break the test unnecessarily if that every changes. If it doesn't matter where these values come from, you could probably get away with just putting `{{.+}}` instead of naming the temp variables explicitly.

Reply:
Yeah, I didn't want to be too sensitive to the ExportVerilog style. Good point about * vs +.